### PR TITLE
Centralise serialisation of mcl types

### PR DESCRIPTION
**Pull request by jinmannwong**
Define serialisation/deserialisation of mcl types in mcl_dkg with exception throw if the string can not be mapped to the appropriate. Note, mcl types in block entropy have been left as strings as comparision operator required for map keys have not been defined for mcl public key - own definition can be added in a later PR.

### CODE REVIEW

File: libs/beacon/src/beacon_manager.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ std::vector<BeaconManager::Coefficient> BeaconManager::GetCoefficients()
   std::vector<Coefficient> coefficients;
   for (uint32_t k = 0; k <= polynomial_degree_; k++)
   {
-    coefficients.push_back(C_ik[cabinet_index_][k].getStr());
+    coefficients.push_back(C_ik[cabinet_index_][k]);
```
Comment (ejfitzgerald) [inline libs/beacon/src/beacon_manager.cpp]:
```suggestion
    coefficients.emplace_back(C_ik[cabinet_index_][k]);
```

**Review by ejfitzgerald — CHANGES_REQUESTED** [review]
Just a minor suggestion, but otherwise looks good to me 😁

**Review by ejfitzgerald — DISMISSED** [review]

File: libs/beacon/src/beacon_manager.cpp [hunk 2]
```diff
@@ -654,10 +596,20 @@ bool BeaconManager::Verify(byte_array::ConstByteArray const &group_public_key,
   }
 
   PublicKey tmp;
-  tmp.setStr(std::string(group_public_key));
-
   Signature tmp2;
-  tmp2.setStr(std::string(signature));
+
```
Comment (n-hutton) [inline libs/beacon/src/beacon_manager.cpp:600]: 👍

File: libs/crypto/include/crypto/mcl_dkg.hpp [hunk 3]
```diff
@@ -152,4 +153,96 @@ std::vector<DkgKeyInformation> TrustedDealerGenerateKeys(uint32_t committee_size
 
 }  // namespace mcl
 }  // namespace crypto
+
+namespace serializers {
+template <typename D>
+struct ArraySerializer<crypto::mcl::Signature, D>
```
Comment (n-hutton) [inline libs/crypto/include/crypto/mcl_dkg.hpp:159]: Side note: Signature here refers to the group signature right? Is it worth renaming this type to GroupSignature?

**Review by n-hutton — APPROVED** [review]
LGTM

File: libs/crypto/include/crypto/mcl_dkg.hpp (hunk 3 above)
Comment (jinmannwong) [reply libs/crypto/include/crypto/mcl_dkg.hpp:159]: It is for signature shares and the group signature as they are both of the same type.

**Pull request merged**